packages/dashboard-frontend/src/containers/FactoryLoader/updateDevfileMetadata.ts

```diff
@@ -22,8 +22,8 @@ import getRandomString from '../../services/helpers/random';
 export type FactorySource = { factory?: { params: string } };
 
 export default function updateDevfileMetadata(devfile: api.che.workspace.devfile.Devfile, factoryParams: string, createPolicy: CreatePolicy): api.che.workspace.devfile.Devfile {
-  if (isDevfileV2(devfile)) {
-    const metadata = devfile.metadata;
+  const metadata = devfile.metadata;
+  if (isDevfileV2(devfile) && metadata) {
```

Comment:
well, should not metadata be created if it's not there?

Reply:
I have removed it